actors/builtin/miner/miner_actor.go

```diff
@@ -384,8 +385,11 @@ func (a Actor) SubmitWindowedPoSt(rt Runtime, params *SubmitWindowedPoStParams)
 
 		// Note: We could delay this charge until end of deadline, but that would require more accounting state.
 		totalPenaltyTarget := big.Add(undeclaredPenaltyTarget, declaredPenaltyTarget)
-		penaltyTotal, err = st.UnlockUnvestedFunds(store, currEpoch, totalPenaltyTarget)
+		availableBalance := st.GetUnlockedBalance(rt.CurrentBalance())
```

Comment:
Blocking: don't confuse the naming of the concepts "available" and "unlocked". I had to re-read this whole PR after I got to the state file and realised these were critically distinguished.

Not blocking, but I think it would be better for `st.PenalizeFundsInPriorityOrder` to accept the `rt.CurrentBalance()` as a parameter rather than this value that is itself a function of state, which would make this whole naming confusion go away.

Reply:
If it accepts `rt.CurrentBalance()` it will also have to accept a parameter for accumulated deductions from unlocked funds because rt.CurrentBalance won't update until the funds are burnt after the state transaction.  Maybe that would be clearer but it won't get rid of the need to keep track of the unlocked funds.